Celeste.Mod.mm/Patches/Level.cs

```diff
@@ -583,13 +583,26 @@ private static void _GCCollect() {
         public override void End() {
             orig_End();
 
-            // reload the vanilla Portraits.xml in case it was overridden by a map
-            // else, if the Portraits.xml doesn't have portrait_madeline defined,
-            // the game will crash when trying to load a save file (since it shows madeline's portrait)
-            GFX.PortraitsSpriteBank = new SpriteBank(GFX.Portraits, Path.Combine("Graphics", "Portraits.xml"));
+            Scene nextScene = patch_Engine.NextScene;
+
+            // reload the vanilla Portraits.xml when exiting; if a map overrides the Portraits.xml
+            // and doesn't have portrait_madeline defined, the game would crash when trying to load a save file
+            // (since it shows madeline's portrait)
+            //
+            // however we need to pay attention to the new scene, else we'll reload vanilla portraits too soon
+            // and make custom portraits stop working. therefore, we ignore these scenes as they don't actually
+            // exit the map:
+            // - MapEditor - debug map
+            // - AssetReloadHelper - f5 bind
+            // - LevelLoader - vanilla f1-f3 binds + everest softlock prevention
+            // - LevelExit (if Mode is Restart or GoldenBerryRestart) - restarts the level, duh
+            if (nextScene is not (Editor.MapEditor or AssetReloadHelper or LevelLoader
+                or patch_LevelExit { Mode: LevelExit.Mode.Restart or LevelExit.Mode.GoldenBerryRestart })) {
```

Comment:
`nextScene is not LevelLoader` is equivalent code here, just to handle the ARH since it works wildly differently from other scenes and will call `Scene.End` after the new `LevelLoader` instance has been created. Other scenes ultimately call `new LevelLoader` to switch back to the level, so they're redundant.
Also `Pico8.Emulator` and `OverworldReflectionsFall` do *not* make a `LevelLoader` to go back, so those should be excluded as well, since `Level.End()` is called.
```suggestion
            if (nextScene is not (LevelLoader or Pico8.Emulator or OverworldReflectionsFall)) {
```

This should be enough to handle all everest and vanilla scenes, how to handle modded scenes is still an open question.

Reply:
Suggestion applied. I guess when it comes to modded scenes, they should create a new `LevelLoader` scene which automatically reloads the portraits.